pyfr/plugins/integrate.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+
+import re
+
+import numpy as np
+
+from pyfr.inifile import Inifile
+from pyfr.mpiutil import get_comm_rank_root, get_mpi
+from pyfr.plugins.base import BasePlugin, init_csv
+from pyfr.quadrules import get_quadrule
+from pyfr.nputil import npeval
+
+class IntegratePlugin(BasePlugin):
+    name = 'integrate'
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    def __init__(self, intg, cfgsect, suffix=None):
+        super().__init__(intg, cfgsect, suffix)
+
+        comm, rank, root = get_comm_rank_root()
+        
+        # Underlying elements class
+        self.elementscls = intg.system.elementscls
+
+        # Expressions to integrate
+        c = self.cfg.items_as('constants', float)
+        self.exprs = [(k, self.cfg.getexpr(cfgsect, k, subs=c))
```

Comment:
Is k needed here?

Reply:
This is taken directly from `tavg.py`

Reply:
`tavg.py` stores `k` as it needs it to generate the `fields` key for the `stats` file (see https://github.com/PyFR/PyFR/blob/develop/pyfr/plugins/tavg.py#L145).  However, this is not something which the integration plugin needs to bother itself with as it's not writing a `.pyfrs` type file.